cmd/autoupdate/main.go

```diff
@@ -76,23 +94,51 @@ func main() {
 	// Create http server.
 	listenAddr := getEnv("AUTOUPDATE_HOST", "") + ":" + getEnv("AUTOUPDATE_PORT", "8002")
 	srv := &http.Server{Addr: listenAddr, Handler: mux}
-	if err != nil {
-		log.Fatalf("Can not create tls listener: %v", err)
-	}
 
+	wait := make(chan error)
 	go func() {
-		fmt.Printf("Listen on %s\n", listenAddr)
-		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
-			log.Fatalf("HTTP Server failed: %v", err)
+		waitForShutdown()
+
+		close(closed)
+		if err := srv.Shutdown(context.Background()); err != nil {
+			wait <- fmt.Errorf("HTTP server shutdown: %w", err)
+			return
 		}
+		wait <- nil
 	}()
 
-	waitForShutdown()
+	fmt.Printf("Listen on %s\n", listenAddr)
+	if err := srv.ListenAndServe(); err != http.ErrServerClosed {
+		return fmt.Errorf("HTTP Server failed: %v", err)
+	}
+
+	return <-wait
+}
+
+func secredKey() (string, error) {
```

Comment:
First, you have a spelling mistake all over the code, it is `secret` ;)

This function does not work correctly - it returns `DJANGO_SECRET_KEY='development'` instead of `development`. Please adjust the regex, so that single or double ticks are accepted (`'` or `"`) - compatibility with the deployment. Since this is bash syntax, I cannot make sure, that single ticks are always used.

Reply:
Both is done.

What is also supported is a mix of single and double quotes `DJANGO_SECRET_KEY='development"`. But this should not be a problem.